packages/@aws-cdk/aws-scheduler-targets-alpha/test/sns-publish.test.ts

```diff
@@ -0,0 +1,535 @@
+          },
+          RoleArn: 'arn:aws:iam::123456789012:role/my-role',
+          RetryPolicy: {},
+        },
+      },
+    });
+
+    template.hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'sns:Publish',
+            Effect: 'Allow',
+            Resource: {
+              Ref: 'TopicBFC7AF6E',
+            },
+          },
+        ],
+      },
+      Roles: ['my-role'],
+    });
+  });
+
+  test('creates IAM policy for imported sns topic with imported IAM role in the same account', () => {
+    const importedTopic = sns.Topic.fromTopicArn(stack, 'ImportedTopic', 'arn:aws:sns:us-east-1:123456789012:my-topic');
+    const importedRole = iam.Role.fromRoleArn(stack, 'ImportedRole', 'arn:aws:iam::123456789012:role/my-role');
+
+    const target = new SnsPublish(importedTopic, {
+      role: importedRole,
+    });
+
+    new scheduler.Schedule(stack, 'Schedule', {
+      schedule: scheduleExpression,
+      target,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResource('AWS::Scheduler::Schedule', {
+      Properties: {
+        Target: {
+          Arn: 'arn:aws:sns:us-east-1:123456789012:my-topic',
+          RoleArn: 'arn:aws:iam::123456789012:role/my-role',
+          RetryPolicy: {},
+        },
+      },
+    });
+
+    template.hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'sns:Publish',
+            Effect: 'Allow',
+            Resource: 'arn:aws:sns:us-east-1:123456789012:my-topic',
+          },
+        ],
+      },
+      Roles: ['my-role'],
+    });
+  });
+
+  test('throws when sns topic is imported from different account', () => {
+    const differentAccountStack = new Stack(app, 'DummyStack', {
+      env: {
+        account: '234567890123',
+        region: 'us-east-1',
+      },
+    });
+
+    const differentAccountTopic = new sns.Topic(differentAccountStack, 'DummyTopic', {
+      topicName: 'DummyTopic',
+    });
+    const target = new SnsPublish(differentAccountTopic, {});
+
+    expect(() =>
+      new scheduler.Schedule(stack, 'Schedule', {
+        schedule: scheduleExpression,
+        target,
+      })).toThrow(/Both the schedule and the topic must be in the same account/);
+  });
+
+  test('throws when sns topic is imported from different region', () => {
+    const differentRegionStack = new Stack(app, 'DummyStack', {
+      env: {
+        account: '123456789012',
+        region: 'us-west-2',
+      },
+    });
+
+    const differentRegionTopic = new sns.Topic(differentRegionStack, 'DummyTopic', {
+      topicName: 'DummyTopic',
+    });
+    const target = new SnsPublish(differentRegionTopic, {});
+
+    expect(() =>
+      new scheduler.Schedule(stack, 'Schedule', {
+        schedule: scheduleExpression,
+        target,
+      })).toThrow(/Both the schedule and the topic must be in the same region/);
+  });
```

Comment:
```suggestion
  test.each([
    ['account', 'arn:aws:sns:us-east-1:999999999999:topic', /Both the schedule and the topic must be in the same account./],
    ['region', 'arn:aws:sns:eu-central-1:123456789012:topic', /Both the schedule and the topic must be in the same region./],
  ])('throws when SNS topic is imported from different %s', (_, arn: string, expectedError: RegExp) => {
    const importedSnsTopic = Topic.fromTopicArn(stack, 'ImportedTopic', arn);
    const target = new SnsPublish(importedSnsTopic, {});

    expect(() =>
      new Schedule(stack, 'MyScheduleDummy', {
        schedule: expr,
        target: target,
      })).toThrow(expectedError);
  });
```
I think that the logic should be kept consistent with the one used in the other targets.
We can resolve the environment in the [`fromTopicArn`](https://github.com/aws/aws-cdk/blob/7c497ee998976b69ad1e2cd3e68f30cb593f8ded/packages/aws-cdk-lib/aws-sns/lib/topic.ts#L63) method directly:
```
  public static fromTopicArn(scope: Construct, id: string, topicArn: string): ITopic {
    const parsedArn = Stack.of(scope).splitArn(topicArn, ArnFormat.SLASH_RESOURCE_NAME);

    class Import extends TopicBase {
      ...
    }

    return new Import(scope, id, {
      account: parsedArn.account,
      region: parsedArn.region,
    });
  }
```

Reply:
You have a good point! I used different logic from the other targets since `.fromTopicArn` doesn't work well with different regions or accounts.

Thanks for your helpful suggestion. I accept it and add unit tests for `sns.test.ts` for `.fromTopicArn`